identity/tests/rpc_endpoints.rs

```diff
@@ -0,0 +1,515 @@
+
+        // create new database
+        diesel::sql_query(format!("CREATE DATABASE \"{}\"", db_name))
+            .execute(&conn)
+            .unwrap_or_else(|_| panic!("Could not create database: {}", db_name));
+
+        // reconnect to the new database
+        let conn = PgConnection::establish(&format!("{}/{}", connection_url, db_name))
+            .expect("Could not connect to database service");
+
+        // run migration
+        embedded_migrations::run(&conn).expect("Failed to apply database migration");
+
+        // insert example data
+        let path = format!("{}/{}", env!("CARGO_MANIFEST_DIR"), "/tests/test_data.sql");
+        match File::open(path) {
+            Ok(file) => {
+                let buffer = BufReader::new(file);
+                for statement in buffer.split(b';') {
+                    let query = String::from_utf8(statement.unwrap())
+                        .unwrap()
+                        .replace("\n", "");
+                    if query.is_empty() {
+                        continue;
+                    };
+                    diesel::sql_query(query)
+                        .execute(&conn)
+                        .expect("Could not insert example data");
+                }
+            }
+            Err(_) => panic!("Could not find test data file"),
+        };
+
+        Self {
+            connection_url,
+            db_name,
+        }
+    }
+
+    fn get_connection_url(&self) -> String {
+        format!("{}/{}", self.connection_url, self.db_name)
+    }
+}
+
+impl Drop for DbTestContext {
+    fn drop(&mut self) {
+        let conn = PgConnection::establish(&self.connection_url)
+            .expect("Could not connect to database service");
+
+        diesel::sql_query(format!(
+            "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = '{}'",
+            self.db_name
+        ))
+        .execute(&conn)
+        .unwrap();
+
+        diesel::sql_query(format!("DROP DATABASE \"{}\"", self.db_name))
+            .execute(&conn)
+            .unwrap_or_else(|_| panic!("Could not drop database: {}", self.db_name));
+    }
+}
+
+pub fn get_test_configuration() -> Configuration {
+    std::env::set_var("RPC_HOST_PORT", "0");
+    std::env::set_var("OAUTH_CLIENT_IDENTIFIER", "test_oauth_client_identifier");
+    std::env::set_var("OAUTH_CLIENT_SECRET", "test_oauth_client_secret");
+    std::env::set_var("JWT_SECRET", "test_jwt_secret");
+
+    Configuration::init_from_env().unwrap()
+}
+
+pub async fn get_db_connection(connection_url: String) -> Pool<ConnectionManager<PgConnection>> {
+    Pool::new(ConnectionManager::new(connection_url)).unwrap()
+}
+
+pub async fn identity_client(
+    addr: SocketAddr,
+) -> std::io::Result<identity::rpc::service::IdentityServiceClient> {
+    identity::rpc::service::IdentityServiceClient::new(
+        Config::default(),
+        tarpc::serde_transport::tcp::connect(addr, Json::default).await?,
+    )
+    .spawn()
+}
```

Comment:
Can't we import this from the api? Or provide it in this crate and import it into the api?

Reply:
I already planned to move it in this crate. This was just a quick and dirty implementation. :-D